### PR TITLE
Instrumentation: Add dashboard uid to traces

### LINKED ISSUE #17410: Instrumentation: Add dashboard UID to traces.
The Graphite and Prometheus datasource adds the dashboard ID to OpenTracing. Since we introduced UID for dashboards in Grafana we should add that to tracing as well.

### PR DESCRIPTION
**Pull request by bergquist**
closes #17410

### CODE REVIEW

File: packages/grafana-ui/src/types/datasource.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import { ComponentType, ComponentClass } from 'react';
-import { TimeRange } from './time';
-import { PluginMeta, GrafanaPlugin } from './plugin';
-import { TableData, TimeSeries, SeriesData, LoadingState } from './data';
-import { PanelData } from './panel';
+import { ComponentClass, ComponentType } from 'react';
```
Comment (bergquist) [inline packages/grafana-ui/src/types/datasource.ts]: not sure why my vscode rearranges the imports... is this desired?

**Comment by bergquist** [comment]
sorry for the force push. I resolved it with rebase instead of merging master into the branch... It's hard to get rid of old practices.

File: public/app/features/dashboard/state/PanelQueryRunner.ts [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
+  TimeRange,
 } from '@grafana/ui';
+import kbn from 'app/core/utils/kbn';
+// Services & Utils
+import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
+import templateSrv from 'app/features/templating/template_srv';
+import cloneDeep from 'lodash/cloneDeep';
+import throttle from 'lodash/throttle';
+import { PartialObserver, Subject, Unsubscribable } from 'rxjs';
+import { PanelQueryState } from './PanelQueryState';
```
Comment (torkelo) [inline public/app/features/dashboard/state/PanelQueryRunner.ts]: can you remove the plugin that scrambles the import order? This make the imports all messy

File: packages/grafana-ui/src/types/datasource.ts (hunk 1 above)
Comment (torkelo) [reply packages/grafana-ui/src/types/datasource.ts]: Does not happen for others, maybe a setting that is on for you? or some plugin?

File: public/app/plugins/datasource/graphite/datasource.ts [hunk 3]
```diff
@@ -66,6 +66,7 @@ export function GraphiteDatasource(this: any, instanceSettings, $q, backendSrv,
     const proxyMode = !this.url.match(/^http/);
     if (proxyMode) {
       httpOptions.headers['X-Dashboard-Id'] = options.dashboardId;
+      httpOptions.headers['X-Dashboard-UId'] = options.dashboardUId;
```
Comment (torkelo) [inline public/app/plugins/datasource/graphite/datasource.ts]: I think we can do this more generically inside https://github.com/grafana/grafana/blob/master/public/app/core/services/backend_srv.ts#L151 . 

From there figure out what dashboard is open add these headers in one location for all data source requests

**Comment by bergquist** [comment]
Closing this and opening a new PR instead.

**Comment by marefr** [comment]
@bergquist you never opened a new PR for this right?

**Pull request closed without merging**